lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3993,29 +3994,32 @@ bool SPIRVToLLVM::transOCLMetadata(SPIRVFunction *BF) {
                                  return MDString::get(*Context, Qual);
                                });
   // Generate metadata for kernel_arg_type
-  if (!transKernelArgTypeMedataFromString(Context, BM, F))
+  if (!transKernelArgTypeMedataFromString(Context, BM, F,
```

Comment:
This nonstandard metadata should only be consumed by the translator when the metadata preservation flag is enabled. The code to ensure this doesn't seem to have been added. Have I understood correctly? If so, please conditionalize this accordingly.

Reply:
Looks good.